provider/pkg/schema/extension_resource.go

```diff
@@ -19,6 +32,169 @@ func (e *ExtensionResource) ToSpec() CloudAPIResource {
 	}
 }
 
+func (e *ExtensionResource) ToOutputs(outputs map[string]any) resource.PropertyMap {
+	return resource.NewPropertyMapFromMap(map[string]any{
+		"type":         e.Type,
+		"properties":   e.Properties,
+		"writeOnly":    e.WriteOnly,
+		"tagsProperty": e.TagsProperty,
+		"tagsStyle":    e.TagsStyle,
+		"outputs":      outputs,
+	})
+}
+
+type extensionResource struct {
+	host   *provider.HostClient
+	client client.CloudControlClient
+}
+
+func NewExtensionResource(host *provider.HostClient, client client.CloudControlClient) *extensionResource {
+	return &extensionResource{
+		host:   host,
+		client: client,
+	}
+}
+
+func (r *extensionResource) Create(ctx context.Context, urn resource.URN, inputs resource.PropertyMap) (identifier *string, outputs resource.PropertyMap, err error) {
+	var extensionResource ExtensionResource
+	_, err = resourcex.Unmarshal(&extensionResource, inputs, resourcex.UnmarshalOptions{})
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to unmarshal inputs: %w", err)
+	}
+
+	id, resourceState, err := r.client.Create(ctx, extensionResource.Type, extensionResource.Properties)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create resource: %w", err)
+	}
+
+	return id, extensionResource.ToOutputs(resourceState), nil
+}
+
+func (r *extensionResource) Read(ctx context.Context, urn resource.URN, id string, oldInputs, oldState resource.PropertyMap) (resource.PropertyMap, resource.PropertyMap, bool, error) {
```

Comment:
This entire method needs better docs and tests. I'm mostly skipping its logic for now in my review.

Reply:
Agree, at some point unit tests would be nice, but this PR focuses on just moving the provider from a point of the resource being unusable to it passing a basic integration test. I would suggest we line up some follow-up issues for testing.

Reply:
See my other comment; I don't agree it can be a follow-up